cpp-package/example/inference/simple_rnn.cpp

```diff
@@ -0,0 +1,359 @@
+ *
+ * Ensure to use the same dictionary file that was used for training the network.
+ */
+void Predictor::LoadDictionary(const std::string& input_dictionary) {
+  if (!FileExists(input_dictionary)) {
+    LG << "Dictionary file " << input_dictionary << " does not exist";
+    throw std::runtime_error("Dictionary file does not exist");
+  }
+  LG << "Loading the dictionary file.";
+  std::ifstream fi(input_dictionary.c_str());
+  if (!fi.is_open()) {
+    std::cerr << "Error opening dictionary file " << input_dictionary << std::endl;
+    assert(false);
+  }
+  std::string line;
+  std::string word;
+  int index;
+  while (std::getline(fi, line)) {
+    std::istringstream stringline(line);
+    stringline >> word >> index;
+    wordToInt[word] = index;
+    intToWord[index] = word;
+  }
+  fi.close();
+}
+
+
+/*
+ * The function populates the input vector with indices from dictionary that are
+ * corresponding to the words in the input string.
+ */
+void Predictor::ConverToIndexVector(const std::string& input, std::vector<float> *input_vector) {
+  std::istringstream input_string(input);
+  input_vector->clear();
+  char delimiter = ' ';
+  std::string token;
+  int words = 0;
+  while (std::getline(input_string, token, delimiter) && (words <= input_vector->size())) {
+    input_vector->push_back(static_cast<float>(wordToInt[token]));
+    words++;
+  }
+  return;
+}
+
+
+/*
+ * The following function runs the forward pass on the model.
+ * The executor is created in the constructor.
+ */
+void Predictor::PredictText(const std::string& input_text) {
+  /*
+   * Initialize a vector of length equal to 'sequence_lenght' with 0.
+   * Convert the input string to a vector of indices that represent
+   * the words in the input string.
+   */
+  std::vector<float> array(sequence_length, 0);
+  ConverToIndexVector(input_text, &array);
+
+  Shape input_shape(sequence_length, 1);
+  NDArray input_data = NDArray(input_shape, global_ctx, false);
+  input_data.SyncCopyFromCPU(array.data(), input_shape.Size());
+  NDArray::WaitAll();
+
+  input_data.CopyTo(&(executor->arg_dict()["data"]));
+  NDArray::WaitAll();
+
+  // Run the forward pass.
+  executor->Forward(false);
+
+  // The output is available in executor->outputs.
+  std::vector<NDArray> outputs = executor->outputs;
+  auto arrayout = executor->outputs[4].Copy(global_ctx);
+  NDArray::WaitAll();
+
+  /*
+   * The output is reshaped to [sequence_length, vocab_size]
+   * The vocab_size in this case is equal to the size of dictionary.
+   * The output will contain the probability distribution for each of the
+   * word in sequence.
+   * We will run ArgmaxChannel operator to find out the index with the
+   * highest probability. This index will point to word in the predicted
+   * output string.
+   */
+  arrayout = arrayout.Reshape(Shape(sequence_length, -1));
+  arrayout = arrayout.ArgmaxChannel();
+  NDArray::WaitAll();
+
+  std::ostringstream oss;
```

Comment:
As I mentioned in other comments, I suggest you consider that Predictor returns the string and the caller function prints it.

Reply:
Modified the PredictText method to return the string.